homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,186 @@
+
+            return await self.async_step_pairing()
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_pairing(self, user_input=None):
+        """Display pairing form."""
+        errors = {}
+        if user_input is not None:
+            try:
+                client = await async_control_connect(self._user_input[CONF_HOST], None)
+            except PyLGTVPairException:
+                return self.async_abort(reason="error_pairing")
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            else:
+                return await self.async_step_register(self._user_input, client)
+
+        return self.async_show_form(
+            step_id="pairing",
+            errors=errors,
+        )
+
+    async def async_step_register(self, user_input, client=None):
+        """Register entity."""
+        if client.is_registered():
+            if client.client_key is None:
+                return self.async_abort(reason="client_key_not_found")
+
+            await self.async_set_unique_id(client.software_info["device_id"])
+            self._abort_if_unique_id_configured()
+
+            data = {
+                CONF_HOST: self._user_input[CONF_HOST],
+                CONF_NAME: self._user_input[CONF_NAME],
+                CONF_CLIENT_SECRET: client.client_key,
+            }
+            return self.async_create_entry(
+                title=user_input.get(CONF_NAME, CONF_HOST), data=data
+            )
+
+        return await self.async_step_user()
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered Webostv device."""
+        user_input = {
+            CONF_HOST: urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname,
+            CONF_NAME: discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME],
+        }
+        self.context["title_placeholders"] = {"name": user_input[CONF_NAME]}
+        self._force_pairing = True
+        return await self.async_step_user(user_input)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.options = config_entry.options
+        self.host = config_entry.data[CONF_HOST]
+        self.key = config_entry.data[CONF_CLIENT_SECRET]
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        errors = {}
+        if user_input is not None:
+            if user_input.get(CONF_ON_ACTION) and not self.hass.states.get(
+                user_input.get(CONF_ON_ACTION)
+            ):
+                errors["base"] = "script_notfound"
+            elif (
+                user_input.get(CONF_ON_ACTION)
+                and (self.hass.states.get(user_input.get(CONF_ON_ACTION))).domain
+                != "script"
+            ):
```

Comment:
Can we avoid multiple lookups to the state machine ?

Reply:
This whole block was too complicated, rewrote it in a simpler way